arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -121,12 +123,103 @@ def run_timerlat(
         if self.finished_early:
             timerlat_proc.send_signal(2)
 
-        if params.enable_time_series and trace_file_exists:
+        # Example time series output from the tracer (truncated)
+        # <idle>-0      [009] d. 123.769498: #1002 context    irq timer_latency  458 ns
+        #  <...>-625890 [009] .. 123.769499: #1002 context thread timer_latency 666 ns
+        #  <...>-625890 [009] .. 123.769500: #1002 context user-ret timer_latency 499 ns
+        # <idle>-0      [002] d. 123.769528: #1003 context    irq timer_latency  587 ns
+        #  <...>-625883 [002] .. 123.769532: #1003 context thread timer_latency 712 ns
+        #  <...>-625883 [002] .. 123.769534: #1003 context user-ret timer_latency 462 ns
+
+        if params.enable_time_series and trace_file:
             # Interrupt the time series collection process and capture the output
             timeseries_proc.send_signal(2)
-            timeseries_output, _ = timeseries_proc.communicate()
+            timeseries_file.close()
+            trace_file.close()
+
+            # The calculation of the offset from uptime to current time certainly means
+            # that our time series is not 100% accurately aligned to the nanosecond, but
+            # the relative times will be accurate. We'll accept this as good enough.
+            uptime_offset = time.time() - time.monotonic()
+
+            with open("./timeseries_file") as timeseries_output:
+
+                if all(False for _ in timeseries_output):
+                    print(
+                        "No results reading tracer output; "
+                        "Skipping time series collection\n"
+                    )
+
+                for line in timeseries_output:
+                    line_list = line.split()
+
+                    # The first object in the line is a string, and it is possible for
+                    # it to have spaces in it. We'll try to cast the second object,
+                    # which should be the CPU number surrounded with [], to an int. If
+                    # that fails, we discard that object from the line_list.
+                    for i, _ in enumerate(line_list):
+                        if i == 1:
+                            try:
+                                cpu_val = re.compile(r"\[([0-9][0-9][0-9])\]")
+                                cpu = int(cpu_val.match(line_list[i]).group(1))
+                                break
+                            except (ValueError, AttributeError):
+                                line_list.pop(i)
+                            except IndexError as error:
+                                print(
+                                    "Unable to determine CPU number; Skipping time "
+                                    f"series collection: {error}\n{line}"
+                                )
+                                timeseries_dict = {}
+                                break
```

Comment:
If what you're trying to do is to find the first occurrence in the line of a number surrounded by square brackets, I think you can dispense with the loop and do that with just the regex....
```suggestion
                    # Find the first occurrence in the line of a three-digit number
                    # surrounded by square brackets. 
                    cpu_val = re.compile(r"^[^[]*\[([0-9]{3})\]")
                    matches = cpu_val.match(line)
                    if matches:
                        cpu = int(matches.group(1))
                    else:
                            print("Unable to determine CPU number; Skipping time "
                                f"series collection:\n\t{line}")
                            timeseries_dict = {}
```
Note that the `break` at line 174 leaves the loop at line 160, and not the loop at line 153, so, as things stand, you're gonna run into a problem at line 199 because `cpu` won't have been (re)assigned...so you need to fix this code _somehow_.

Reply:
The problem with this is that I am counting on the `.pop()` re-indexing the the `line_list` for the below `try` where I assign the other variables from the fields of the line.

Reply:
I've posted another, [more comprehensive suggestion](https://github.com/arcalot/arcaflow-plugin-rtla/pull/8#discussion_r1866563404), below.